src/state/farms/fetchFarmsPrices.ts

```diff
@@ -0,0 +1,102 @@
+  }
+
+  if (farm.quoteToken.symbol === 'wBNB') {
+    return hasTokenPriceVsQuote ? bnbPriceBusd.times(farm.tokenPriceVsQuote) : BIG_ZERO
+  }
+
+  // We can only calculate profits without a quoteTokenFarm for BUSD/BNB farms
+  if (!quoteTokenFarm) {
+    return BIG_ZERO
+  }
+
+  // Possible alternative farm quoteTokens:
+  // UST (i.e. MIR-UST), pBTC (i.e. PNT-pBTC), BTCB (i.e. bBADGER-BTCB), ETH (i.e. SUSHI-ETH)
+  // If the farm's quote token isn't BUSD or wBNB, we then use the quote token, of the original farm's quote token
+  // i.e. for farm PNT - pBTC we use the pBTC farm's quote token - BNB, (pBTC - BNB)
+  // from the BNB - pBTC price, we can calculate the PNT - BUSD price
+  if (quoteTokenFarm.quoteToken.symbol === 'wBNB') {
+    const quoteTokenInBusd = bnbPriceBusd.times(quoteTokenFarm.tokenPriceVsQuote)
+    return hasTokenPriceVsQuote && quoteTokenInBusd
+      ? new BigNumber(farm.tokenPriceVsQuote).times(quoteTokenInBusd)
+      : BIG_ZERO
+  }
+
+  if (quoteTokenFarm.quoteToken.symbol === 'BUSD') {
+    const quoteTokenInBusd = quoteTokenFarm.tokenPriceVsQuote
+    return hasTokenPriceVsQuote && quoteTokenInBusd
+      ? new BigNumber(farm.tokenPriceVsQuote).times(quoteTokenInBusd)
+      : BIG_ZERO
+  }
+
+  // Catch in case token does not have immediate or once-removed BUSD/wBNB quoteToken
+  return BIG_ZERO
+}
+
+const getFarmQuoteTokenPrice = (farm: Farm, quoteTokenFarm: Farm, bnbPriceBusd: BigNumber): BigNumber => {
+  if (farm.quoteToken.symbol === 'BUSD') {
+    return new BigNumber(1)
+  }
+
+  if (farm.quoteToken.symbol === 'wBNB') {
+    return bnbPriceBusd
+  }
+
+  if (!quoteTokenFarm) {
+    return BIG_ZERO
+  }
+
+  if (quoteTokenFarm.quoteToken.symbol === 'wBNB') {
+    return quoteTokenFarm.tokenPriceVsQuote ? bnbPriceBusd.times(quoteTokenFarm.tokenPriceVsQuote) : BIG_ZERO
+  }
+
+  if (quoteTokenFarm.quoteToken.symbol === 'BUSD') {
+    return quoteTokenFarm.tokenPriceVsQuote ? new BigNumber(quoteTokenFarm.tokenPriceVsQuote) : BIG_ZERO
+  }
+
+  return BIG_ZERO
+}
+
+const fetchFarmsPrices = async (farms, apiPriceData) => {
+  const bnbBusdFarm = farms.find((farm: Farm) => farm.pid === 252)
+  const bnbPriceBusd = bnbBusdFarm.tokenPriceVsQuote ? new BigNumber(1).div(bnbBusdFarm.tokenPriceVsQuote) : BIG_ZERO
+
+  const farmsWithPrices = farms.map((farm) => {
+    const quoteTokenFarm = getFarmFromTokenSymbol(farms, farm.quoteToken.symbol)
+    let baseTokenPrice = getFarmBaseTokenPrice(farm, quoteTokenFarm, bnbPriceBusd)
+    let quoteTokenPrice = getFarmQuoteTokenPrice(farm, quoteTokenFarm, bnbPriceBusd)
+
+    // Catches to use API price if price cannot be obtained from farm. As of 24.5.21 this is only KUN-QSD
+    if (!baseTokenPrice.gt(0)) {
+      const tokenAddress = getAddress(farm.token.address)
+      baseTokenPrice = apiPriceData ? new BigNumber(apiPriceData[tokenAddress.toLowerCase()]) : BIG_ZERO
+    }
+    if (!quoteTokenPrice.gt(0)) {
```

Comment:
Do these always come back as numbers? Is the default 0? I would prefer we do not invert a check (readability). Can you use `eq`, `lt`, or `lte`?

Reply:
Yea `getFarmBaseTokenPrice` / `getFarmQuoteTokenPrice` will always return BigNumbers. Good shout on making the condition neater, I have added that.